src/game/Pet.cpp

```diff
@@ -90,7 +90,90 @@ void Pet::RemoveFromWorld()
+        result = CharacterDatabase.PQuery("SELECT id, entry, owner, modelid, level, exp, Reactstate, loyaltypoints, loyalty, trainpoint, slot, name, renamed, curhealth, curmana, curhappiness, abdata, TeachSpelldata, savetime, resettalents_cost, resettalents_time, CreatedBySpell, PetType, xpForNextLoyalty "
+            "FROM character_pet WHERE owner = '%u' AND entry = '%u' AND (slot = '%u' OR slot > '%u') ",
+            ownerid, petentry, PET_SAVE_AS_CURRENT, PET_SAVE_LAST_STABLE_SLOT);
+    else
+        // any current or other non-stabled pet (for hunter "call pet")
+        //                                        0   1      2      3        4      5    6           7              8        9           10    11    12       13         14       15            16      17              18        19                 20                 21              22       23
+        result = CharacterDatabase.PQuery("SELECT id, entry, owner, modelid, level, exp, Reactstate, loyaltypoints, loyalty, trainpoint, slot, name, renamed, curhealth, curmana, curhappiness, abdata, TeachSpelldata, savetime, resettalents_cost, resettalents_time, CreatedBySpell, PetType, xpForNextLoyalty "
+            "FROM character_pet WHERE owner = '%u' AND (slot = '%u' OR slot > '%u') ",
+            ownerid, PET_SAVE_AS_CURRENT, PET_SAVE_LAST_STABLE_SLOT);
+
+    if (!result)
+        return SPELL_FAILED_NO_PET;
+
+    Field* fields = result->Fetch();
+
+    // update for case of current pet "slot = 0"
+    petentry = fields[1].GetUInt32();
+    if (!petentry)
+    {
+        delete result;
+        return SPELL_FAILED_NO_PET;
+    }
+
+    CreatureInfo const* creatureInfo = ObjectMgr::GetCreatureTemplate(petentry);
+    if (!creatureInfo)
+    {
+        sLog.outError("Pet entry %u does not exist but used at pet load (owner: %s).", petentry, owner->GetGuidStr().c_str());
+        delete result;
+        return SPELL_FAILED_NO_PET;
+    }
+
+    uint32 summon_spell_id = fields[21].GetUInt32();
+    SpellEntry const* spellInfo = sSpellStore.LookupEntry(summon_spell_id);
+
+    bool is_temporary_summoned = spellInfo && GetSpellDuration(spellInfo) > 0;
+
+    // check temporary summoned pets like mage water elemental
+    if (current && is_temporary_summoned)
+    {
+        delete result;
+        return SPELL_FAILED_NO_PET;
+    }
+
+    PetType pet_type = PetType(fields[22].GetUInt8());
+    if (pet_type == HUNTER_PET)
+    {
+        if (!creatureInfo->isTameable())
+        {
+            delete result;
+            return SPELL_FAILED_NO_PET;
+        }
+    }
+
+    uint32 pet_health = fields[13].GetUInt32();
+    if (!pet_health)
+    {
+        delete result;
+        return SPELL_FAILED_TARGETS_DEAD;
```

Comment:
Yeah, I'm the one who put that one there originally as a placeholder since I couldn't find the right one either, should have probably added a comment in the code about that..